tests/compiler_env_test.py

```diff
@@ -80,5 +80,16 @@ def test_logger_forced():
         env_b.close()
 
 
+def test_set_seed(env: CompilerEnv):
+    assert env.seed(1) == [1]
+    assert env.seed(2) == [2]
+
+
+def test_choose_random_seed(env: CompilerEnv):
+    # Check that at least once a unique seed is generated.
+    seeds = {env.seed()[0] for _ in range(10)}
+    assert len(seeds) > 1
```

Comment:
You should call seed twice and make sure the seeds are different. I had problems in the past since the time() function often returned the same value (may only have had seconds resolution)

Reply:
Good catch, done.

I was never quite sure how precise `time()` was. From what I've read it is "usually" microseconds, except on Windows where it's second resolution. `time_ns()` in Py 3.7 is a nice addition